llama_stack/distribution/server/auth_providers.py

```diff
@@ -339,6 +339,134 @@ async def close(self):
+    claims_mapping: dict[str, str] = Field(
+        default_factory=lambda: {
+            "username": "roles",
+            "groups": "teams",
+            "uid": "roles",
+        },
+        description="Mapping of OpenShift user claims to access attributes",
+    )
+
+    @field_validator("claims_mapping")
+    @classmethod
+    def validate_claims_mapping(cls, v):
+        for key, value in v.items():
+            if not value:
+                raise ValueError(f"claims_mapping value cannot be empty: {key}")
+        return v
+
+
+class OpenShiftOAuthAuthProvider(AuthProvider):
+    """
+    OpenShift OAuth authentication provider that validates tokens using the OpenShift user API.
+    This provider integrates with OpenShift's built-in OAuth server by using the
+    /apis/user.openshift.io/v1/users/~ endpoint to validate tokens and extract user information.
+    """
+
+    def __init__(self, config: OpenShiftOAuthAuthProviderConfig):
+        self.config = config
+
+    async def validate_token(self, token: str, scope: dict | None = None) -> User:
+        """Validate a token using OpenShift's user API endpoint."""
+
+        # Configure SSL context
+        ssl_ctxt = None
+        if self.config.tls_cafile:
+            ssl_ctxt = ssl.create_default_context(cafile=self.config.tls_cafile.as_posix())
+        elif not self.config.verify_tls:
+            ssl_ctxt = ssl.create_default_context()
+            ssl_ctxt.check_hostname = False
+            ssl_ctxt.verify_mode = ssl.CERT_NONE
+
+        # Build the OpenShift user API endpoint URL
+        user_api_url = f"{self.config.api_server_url.rstrip('/')}/apis/user.openshift.io/v1/users/~"
+
+        try:
+            async with httpx.AsyncClient(verify=ssl_ctxt if ssl_ctxt else self.config.verify_tls) as client:
+                response = await client.get(
+                    user_api_url,
+                    headers={
+                        "Authorization": f"Bearer {token}",
+                    },
+                    timeout=10.0,
+                )
+
+                if response.status_code == 401:
+                    raise ValueError("Invalid token")
+                elif response.status_code != 200:
+                    logger.warning(f"OpenShift user API failed with status code: {response.status_code}")
+                    raise ValueError(f"Token validation failed: {response.status_code}")
+
+                user_response = response.json()
+
+                # Extract user information from OpenShift user object
+                metadata = user_response.get("metadata", {})
+                if not metadata:
+                    raise ValueError("No metadata found in user response")
+
+                username = metadata.get("name")
+                if not username:
+                    raise ValueError("No username found in user response")
+
+                uid = metadata.get("uid")
```

Comment:
Is the uid here a uuid as with other kubernetes objects? If so is this field actually useful in this context over the simpler username as retrieved above? Seems like it would be rather unwieldy to use.

Reply:
That's correct, uid is not relevant here.